pkg/skaffold/build/custom/custom.go

```diff
@@ -68,9 +69,26 @@ func (b *ArtifactBuilder) Build(ctx context.Context, out io.Writer, a *latest.Ar
 
 func (b *ArtifactBuilder) retrieveCmd(out io.Writer, a *latest.Artifact, tag string) (*exec.Cmd, error) {
 	artifact := a.CustomArtifact
+
+	var cmd *exec.Cmd
+
 	split := strings.Split(artifact.BuildCommand, " ")
+	switch {
+	// When the user specifies a command without args, such as `./build.sh`,
+	// Skaffold can just exec this command.
+	case len(split) == 1:
+		cmd = exec.Command(artifact.BuildCommand)
+
+	// When the user specifies a command with args, such as `./build.sh -flag $IMAGE`,
+	// Skaffold has to shell out to sh in order for arguments to be evaluated.
```

Comment:
It's possible the executable is an environment variable too.  I define `$JAVA7`, `$JAVA8`, and `$JAVA11` in my environment to invoke the corresponding `java` binary.  Maybe we should just use `sh`/`cmd` always?

Reply:
we could do that

Reply:
I'm up for that. This split based determination is a "poor man's parsing" anyway - a proxy for "does this command have env vars"